compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/platform/BackingTextArea.kt

```diff
@@ -95,62 +100,51 @@ internal class BackingTextArea(
             setProperty("text-shadow", "none")
         }
 
-        htmlInput.addEventListener("input", {
-            val text = htmlInput.value
-            val cursorPosition = htmlInput.selectionEnd
-            sendImeValueToCompose(onEditCommand, text, cursorPosition)
+        htmlInput.addEventListener("input", { evt ->
+            evt.preventDefault()
+            evt as InputEventExtended
+
+            when (evt.inputType) {
+                "insertLineBreak" -> {
+                    if (imeOptions.singleLine) {
+                        onImeActionPerformed(imeOptions.imeAction)
+                    }
+                }
+                "insertCompositionText" -> {
+                    onEditCommand(listOf(SetComposingTextCommand(evt.data!!, 1)))
+                }
+                "insertText" -> {
+                    val data = evt.data ?: return@addEventListener
+                    if ((data.length == 1)) {
+                        sendKey(KeyboardEvent(
+                            "keydown",
+                            KeyboardEventInit(key = data)
+                        ).toComposeEvent())
+                    } else if (data.length > 1) {
+                        onEditCommand(listOf(CommitTextCommand(data, 1)))
+                    }
+                }
+                "deleteContentBackward" -> {
+                    sendKey(KeyboardEvent(
+                        "keydown",
+                        KeyboardEventInit(key = "Backspace", code="Backspace").withKeyCode(8)
```

Comment:
Please apply auto-formatting

Reply:
Do we have a const for "8"?

Reply:
formatted everything and good catch, de-hardcoded the value